pom.xml

```diff
@@ -287,7 +287,7 @@
         <dependency>
             <groupId>org.apache.kafka</groupId>
             <artifactId>connect-api</artifactId>
-            <version>2.3.0</version>
+            <version>2.7.0</version>
```

Comment:
Is this api upgrade necessary for this fix? I'm not catching any obvious new functionality that would make a difference here.

Reply:
I tried to call this out in the description:
> The connect-api dependency is bumped to 2.7.0 in order to leverage improvements in the SimpleHeaderConverter's parsing logic that, unfortunately, make this issue possible in the Snowflake connector.